code/modules/xenoarchaeology/traits/xenoartifact_activators.dm

```diff
@@ -0,0 +1,151 @@
+
+///============
+/// Signal activator, responds to respective signals sent through signallers
+///============
+/datum/xenoartifact_trait/activator/signal
+	label_name = "Signal"
+	label_desc = "Signal: The material recieves radio frequencies and reacts when a matching code is delivered."
+	charge = 25
+	signals = list(XENOA_SIGNAL)
+	flags = BLUESPACE_TRAIT | URANIUM_TRAIT
+
+/datum/xenoartifact_trait/activator/signal/on_init(obj/item/xenoartifact/X)
+	..()
+	X.code = rand(1, 100) //Random code is shared by all signaller traits
+	X.frequency = FREQ_SIGNALER
+	X.set_frequency(X.frequency)
+	X.max_range += 1
+
+/datum/xenoartifact_trait/activator/signal/on_item(obj/item/xenoartifact/X, atom/user, atom/item)
+	if(istype(item, /obj/item/analyzer))
+		to_chat(user, "<span class='info'>The [item.name] displays a signal-input code of [X.code], and frequency [X.frequency].</span>")
+		return TRUE
+	return ..()
+
+/datum/xenoartifact_trait/activator/signal/calculate_charge(datum/source, obj/item/thing, mob/user, atom/target, params)
+	var/obj/item/xenoartifact/X = source
+	X.default_activate(charge, user, target)
+	log_game("[key_name_admin(user)] signalled [X] at [world.time]. [X] located at [X.x] [X.y] [X.z].")
+
+///============
+/// Battery activator, needs a cell to activate
+///============
+/datum/xenoartifact_trait/activator/batteryneed
+	desc = "Charged"
+	label_desc = "Charged: The material has a natural power draw. Supplying any current to this will cause a reaction."
+	charge = 25
+	signals = list(COMSIG_PARENT_ATTACKBY)
+	flags = BLUESPACE_TRAIT | URANIUM_TRAIT
+
+/datum/xenoartifact_trait/activator/batteryneed/on_item(obj/item/xenoartifact/X, atom/user, atom/item)
+	if(istype(item, /obj/item/multitool))
+		to_chat(user, "<span class='info'>The [item.name] displays a draw of [X.charge_req].</span>")
+		return TRUE
+	return ..()
+
+/datum/xenoartifact_trait/activator/batteryneed/calculate_charge(datum/source, obj/item/stock_parts/cell/thing, mob/user, atom/target, params)
+	var/obj/item/xenoartifact/X = source
+	if(istype(thing))
+		return
+	if(thing.use(X.charge_req*10))
+		X.default_activate(charge, user, user)
+
+///============
+/// Weighted activator, picking up activates
+///============
+/datum/xenoartifact_trait/activator/weighted
+	desc = "Weighted"
+	label_desc = "Weighted: The material is weighted and produces a reaction when picked up."
+	charge = 25
+	signals = list(COMSIG_ITEM_PICKUP)
+	blacklist_traits = list(/datum/xenoartifact_trait/minor/dense, /datum/xenoartifact_trait/minor/anchor, /datum/xenoartifact_trait/major/distablizer)
+	flags = BLUESPACE_TRAIT | URANIUM_TRAIT
+
+/datum/xenoartifact_trait/activator/weighted/calculate_charge(datum/source, obj/item/thing, mob/user, atom/target)
+	var/obj/item/clothing/gloves/artifact_pinchers/P = (locate(/obj/item/clothing/gloves/artifact_pinchers) in target.contents || locate(/obj/item/clothing/gloves/artifact_pinchers) in user.contents)
```

Comment:
Shouldn't this specifically check the gloves slot? Pinchers in my pocket doesn't make sense to be safe